security_best_practices/security_best_practices.sp

```diff
@@ -0,0 +1,243 @@
+}
+
+control "tailscale_acl_ssh_admin_roles_assigned" {
+  title       = "Assign admin roles"
+  description = "Assign user roles for managing Tailscale as appropriate, based on job function and for separation of duties. Tailscale provides multiple user roles that restrict who can modify your tailnet's configurations."
+  sql  = <<-EOT
+    with  tailscale_users as (
+      select
+        tailnet_name,
+        count(*) as tailnet_number
+      from
+        tailscale_acl_ssh
+      where
+        users ?| array['group:admin']and users ?| array['group:itadmin']and users ?| array['group:networkadmin']and users ?| array['group:auditor']
+      group by
+        tailnet_name
+    )
+    select
+      t.tailnet_name as resource,
+      case
+        when tu.tailnet_number > 0 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when tu.tailnet_number > 0 then t.tailnet_name || ' has all admin roles assigned.'
+        else t.tailnet_name || ' does not have all admin roles assigned.'
+      end as reason
+    from
+      tailscale_acl_ssh as t
+      left join tailscale_users as tu on t.tailnet_name = tu.tailnet_name
+    group by
+      tu.tailnet_number,
+      t.tailnet_name;
+  EOT
+}
+
+control "tailscale_acl_ssh_check_mode_enabled" {
+  title       = "Use check mode for tailscale SSH"
+  description = "Verify high-risk Tailscale SSH connections with check mode."
+  sql = <<-EOT
+    with  tailscale_users as (
+      select
+        tailnet_name,
+        count(*) as tailnet_number
+      from
+        tailscale_acl_ssh
+      where
+        users ?| array['root'] and action = 'check' and check_period is not null
+      group by
+        tailnet_name
+    )
+    select
+      t.tailnet_name as resource,
+      case
+        when tu.tailnet_number > 0 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when tu.tailnet_number > 0 then t.tailnet_name || ' has check mode enabled.'
+        else t.tailnet_name || ' has check mode disabled.'
+      end as reason
+    from
+      tailscale_acl_ssh as t
+      left join tailscale_users as tu on t.tailnet_name = tu.tailnet_name
+    group by
+      tu.tailnet_number,
+      t.tailnet_name;
+  EOT
+}
+
+control "tailscale_device_authorization_enabled" {
+  title       = "Enable device authorization"
+  description = "New devices can be manually reviewed and approved by an Admin before they can join the network. This can be used to ensure only trusted devices, such as workplace-managed laptops and phones, can access a network."
+  sql = <<-EOT
+    select
+      id as resource,
+      case
+        when authorized = true then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when authorized = true then name || ' is authorized.'
+        else name || ' is unauthorized.'
+      end as reason,
+      tailnet_name
+    from
+      tailscale_device;
+  EOT
+}
+
+control "tailscale_device_key_expire" {
+  title       = "Customize key expiration"
```

Comment:
```suggestion
  title       = "Customize node key expiry"
```

Assuming this is the same control as https://tailscale.com/kb/1196/security-hardening/#customize-node-key-expiry